--- ruby-1.9.3-p327/lib/mkmf.rb	2012-11-29 10:54:04.000000000 +0100
+++ clone/lib/mkmf.rb	2012-11-29 10:55:27.000000000 +0100
@@ -213,19 +213,7 @@
 end
 
 def merge_libs(*libs)
-  libs.inject([]) do |x, y|
-    xy = x & y
-    xn = yn = 0
-    y = y.inject([]) {|ary, e| ary.last == e ? ary : ary << e}
-    y.each_with_index do |v, yi|
-      if xy.include?(v)
-        xi = [x.index(v), xn].max()
-        x[xi, 1] = y[yn..yi]
-        xn, yn = xi + (yi - yn + 1), yi + 1
-      end
-    end
-    x.concat(y[yn..-1] || [])
-  end
+  libs.flatten.uniq
 end
 
 # This is a custom logging module. It generates an mkmf.log file when you
